### PR TITLE
Fix stable-2.5 branch to ansible-core 2.18

### PR DESCRIPTION
**Pull request by KB-perByte**
## Description

- What is being changed?
   fix ansible-core to 2.18 as the stable-2.5 branch need not be tested with anything after 2.18 as it is not supported 
   https://access.redhat.com/support/policy/updates/ansible-automation-platform <ref
   
- Why is this change needed?
  stable-2.5 is pulling in ansible-core 2.19 as of this PR and having unwanted failures due to all the templating updates.\

- How does this change address the issue?

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Test update
- [ ] Refactoring (no functional changes)
- [ ] Development environment change
- [ ] Configuration change

## Self-Review Checklist

- [ ] I have performed a self-review of my code
- [ ] I have added relevant comments to complex code sections
- [ ] I have updated documentation where needed
- [ ] I have considered the security impact of these changes
- [ ] I have considered performance implications
- [ ] I have thought about error handling and edge cases
- [ ] I have tested the changes in my local environment

### Steps to Test
1. 
2. 
3. 

### Required Actions

- [ ] Requires documentation updates
  
- [ ] Requires downstream repository changes
  
- [ ] Requires infrastructure/deployment changes
  
- [ ] Requires coordination with other teams
  
- [ ] Blocked by PR/MR: #XXX

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -92,7 +92,7 @@ jobs:
           python-version: 3.11
 
       - name: Install requirements
-        run: pip3.11 install -r requirements/requirements_dev.txt ansible-core
+        run: pip3.11 install -r requirements/requirements_dev.txt ansible-core==2.17
```
Comment (djdanielsson) [inline .github/workflows/integration.yml:95]: the title says 2.18 but this shows 2.17
